pkgs/development/compilers/ghc/6.10.4.nix

```diff
@@ -9,7 +9,7 @@ stdenv.mkDerivation rec {
   name = "ghc-${version}";
 
   src = fetchurl {
-    url = "${meta.homepage}/dist/${version}/${name}-src.tar.bz2";
+    url = "https://downloads.haskell.org/~ghc/${version}/ghc-${version}-src.tar.bz2";
```

Comment:
`${name}` is a bad shorthand because the name is prefixed for cross compilers.

Reply:
Longterm: don't suffix (I believe it's suffix, not prefix) the target triplet onto ${name}?

Or should I drop this idea that generally any store path _can_ be created equally by either cross or native compilation?

Reply:
@bjornfor so we prefix for `target != host` and suffix for `host != build` :D. The former actually creates a meaningfully different derivation, while the latter ought to just be the implementation detail you want to hide. So don't worry, nothing here gets you farther from that.